### DISCUSSION TITLE
Using transformix and a deformation field to transform an image

### DISCUSSION THREAD

**Discussion by artoews** [SuperElastix/elastix#1016] [Q&A]
Hi, I have a deformation field in numpy array format and want to use Transformix to apply this deformation field to an image. Basically I want to adapt this [python example script](https://github.com/InsightSoftwareConsortium/ITKElastix/blob/main/examples/ITK_Example15_Transformix_image_TranslationTransform.ipynb), replacing the TranslationTransform with a DeformationFieldTransform. How can this be done?

**Comment by N-Dekker** [comment]
@artoews Very good question, thanks! 

First of all, the transform you specify to `itk.TransformixFilter` must be an "ITK transform". In ITK the deformation field transform type is named `itk.DisplacementFieldTransform`.  Before passing it to `itk.TransformixFilter`, you must set the _DisplacementField_ of this `itk.DisplacementFieldTransform`. 

`itk.DisplacementFieldTransform` cannot be passed to `itk.TransformixFilter.SetTransform`, because it cannot be "converted" internally to an elastix/transformix representation. So instead, you should pass it to `itk.TransformixFilter.SetExternalTransform`

See also
- https://github.com/SuperElastix/elastix/issues/874

I'm sorry we did not yet make a Python example for this use case  🤷
